### ISSUE TITLE
Wins page: Fix question marks appearing where apostrophes are expected

### ISSUE THREAD

**Issue by JessicaLucindaCheng** [hackforla/website#2675]
### Overview
As developers, we want to make sure the website displays information correctly so that the information is accurate and our website looks professional. For this issue, we will figure out what is causing the apostrophes to change to question marks in the Google Apps Script and fix the problem. 

<details>
  <summary>Screenshot of problem: Question marks where apostrophes are expected</summary>
  <img width="626" alt="wins-page-question-marks" src="https://user-images.githubusercontent.com/31293603/148660529-ea857b7c-a2d5-41b8-9737-0aef43563d1c.png">
</details>

### Details
The original submission by Janie Chen did not contain question marks in the Wins-form (Responses) Google Sheet. However, the question marks appear on the [Wins page](https://www.hackforla.org/wins/). The problem may come from the Google Apps Script that generates the [_data/external/_wins-data.json](https://github.com/hackforla/website/blob/54819680d2f6ea830887771984e1fbb60da20f3d/_data/external/_wins-data.json) and [_data/external/wins-data.json](https://github.com/hackforla/website/blob/54819680d2f6ea830887771984e1fbb60da20f3d/_data/external/wins-data.json) because the question marks are seen in the json files.

### Action Items
- [x] Request access to the "HfLA website Admin" Google Drive by asking one of the technical lead(s)/merge team member(s)
- [x] Make copies of files in the "HfLA website Admin" Google Drive in your own drive (for making code changes to Apps Script and for testing)
- [x] Figure out what is causing the apostrophes to change to question marks in the Google Apps Script
- [x] Fix the problem in your own copies of the Apps Script and in your own drive
- [x] Check and test the changes in your own drive
- [x] **Before** the changes are to be added to the production script, demo the code changes in your drive to the dev team leads
- [x] Edit the production script with your code changes
- [x] Check and test production script code changes work fine
- [x] Reach out to the dev team leads and ask them to review the changes to the production script code

### Resources/Instructions
- [Google Apps Script overview](https://developers.google.com/apps-script/overview)
- [Wins page](https://www.hackforla.org/wins/)
- [_data/external/_wins-data.json](https://github.com/hackforla/website/blob/54819680d2f6ea830887771984e1fbb60da20f3d/_data/external/_wins-data.json) 
- [_data/external/wins-data.json](https://github.com/hackforla/website/blob/54819680d2f6ea830887771984e1fbb60da20f3d/_data/external/wins-data.json)

**Comment by JessicaLucindaCheng** [comment]
@SAUMILDHANKAR @R-Tomas-Gonzalez

1. Did you want it so that any backend developer can take this issue and would need to just ask a technical lead for access to the wins submission Google Sheet?

2. I don't have access to the wins Google Sheet and don't know what the link for it is. Can one of you add it to the Resources section?

**Comment by JessicaLucindaCheng** [comment]
> @SAUMILDHANKAR @R-Tomas-Gonzalez
> 
> 1. Did you want it so that any backend developer can take this issue and would need to just ask a technical lead for access to the wins submission Google Sheet?
> 
> 2. I don't have access to the wins Google Sheet and don't know what the link for it is. Can one of you add it to the Resources section?

1. Tomas said just tech lead is fine
2. I'm just going to leave out the resource because I don't have access to it nor a link for it.

**Comment by JessicaLucindaCheng** [comment]
@SAUMILDHANKAR Since the files needed to fix this issue are in the protected admin drive, should I restrict this issue to admins/leads that have access to the drive to work on? Or can it be open to other developers and they would just have to ask a lead for access (similar to how Matt wrote issue #2385)?

**Comment by SAUMILDHANKAR** [comment]
@JessicaLucindaCheng It can be open to other developers as well. (like Matt's issue). Thanks for bringing it up.

**Comment by JessicaLucindaCheng** [comment]
@SAUMILDHANKAR Since you are more familiar with Google AppScripts than I am. Could you review this issue for `ready for milestone`? If it's not ready, please provide feedback as to what I need to change. Thank you.

**Comment by SAUMILDHANKAR** [comment]
> @SAUMILDHANKAR Since you are more familiar with Google AppScripts than I am. Could you review this issue for `ready for milestone`? If it's not ready, please provide feedback as to what I need to change. Thank you.

@JessicaLucindaCheng Thanks for writing the issue. My suggestions: 
- We could go for a shorter title like "Fix question mark error on Wins page" or something similar. I remember from our team meetings to make titles succinct. 
- I would also suggest adding the following points as reminders in the action items (this might help avoiding accidental changes to the production script)
    - make copies of files in the admin drive in their own drive (for making code changes to apps script and testing)
    - demo of changes and recommendations to team leads recommended before the changes could be added to production script.

**Comment by JessicaLucindaCheng** [comment]
@SAUMILDHANKAR I really appreciate you taking the time to give me feedback on writing this issue. I made the requested changes, except I left the title more descriptive because I felt being descriptive and clear was more important than being super succinct. (Though I did shorten the title a bit.)

Please put a `ready for milestone` label on if you think it's ready. Otherwise, let me know what else should be changed.

**Comment by ldaws003** [comment]
Availability: Sunday 12:00pm - 5pm, Monday 3:00pm to 6:00pm, Tuesday 9:00am-5:00pm, and Wednesday 4:00pm-8:00pm
ETA: Wednesday 5:00pm

**Comment by ldaws003** [comment]
Progress: Downloaded the files from the drive and will be going through Google App Script docs before trying to find the bug
Blockers: N/A
Availability: Monday 4:00pm to 7:00pm, Wednesday 2:00pm to 5:00pm, Friday: 5:00pm to 8:00pm
ETA: Saturday

**Comment by ldaws003** [comment]
Progress: Read through documentation, will be searching for the bug solution
Blockers: N/A
Availability: Friday 8:00am-5:00pm, Saturday (all day), Sunday (all day)
ETA: Sunday

**Comment by ldaws003** [comment]
Progress: Searching for the bug
Blockers: N/A
Availability: Thursday (all day), Friday 8:00am-5:00pm, Saturday (all day), Sunday (all day)
ETA: Sunday

**Comment by ldaws003** [comment]
Progress: Searching for bug
Blockers: N/A
Availability: Saturday (all day), Sunday (all day)
ETA: Sunday

**Comment by ldaws003** [comment]
Progress: Still trying to find the bug
    Blockers: Can't find documentation on GHRequest and getting errors trying to run app script
    Availability: Saturday (all day), Sunday (all day)
    ETA: Sunday 10:00pm

**Comment by ldaws003** [comment]
Progress: Fixed Bug waiting to demo for dev team leads
Blockers: None
Availability: Saturday (all day), Sunday (all day)
ETA: Sunday 10:00pm

**Comment by ldaws003** [comment]
Progress: Fixed Bug waiting to demo for dev team leads
Blockers: None
Availability: Monday - Friday 1:00pm to 5:00pm
ETA: Thursday 10:00pm

**Comment by jdingeman** [comment]
Developer demonstrated his code and showed that the wins-data.json file replaced the question marks with apostrophes. We double checked with diffchecker.com to make sure there were no unintended removals or additions. The production code was updated but cannot run due to a "bad credentials" message as shown in the log below:
![image](https://user-images.githubusercontent.com/88953806/211939926-e2af3b7c-da7b-43c8-a68c-91e253c11de9.png)

I expect this may be due to the github authorization token expiration which we should try to fix so we can review the Wins page on the published website and see if it is working. Then we can close this issue without PR.

**Comment by ldaws003** [comment]
Progress: Displayed solution, code was successful but there was an unexpected error not on my part, waiting for response.
Blockers: None
Availability: Just waiting
ETA: N/A

**Comment by jdingeman** [comment]
Tried to run the code after the authorization token was updated. I am not entirely sure how the workflow works for the Google Apps Script issues, because it seems that after running this code, it pushed a commit directly into PR #3024, and from the looks of it, it has indeed fixed the issue of question marks showing up where the apostrophes are supposed to be. Since the most recent commit displays the expected outcome, @ldaws003 I think you should be safe to go onto your next issue if you'd like! Nice job! I'll talk with @hackforla/website-merge regarding PR #3024 to see what we need to do about pushing the code into the codebase. It may have a few dependencies that need to be satisfied before we can merge it. 

We'll leave this issue open until we determine the next step.

**Comment by ldaws003** [comment]
> 

OK. For the fix the only line of code that I had to change was line 93. I changed it from this

`const encodedArrayData = Utilities.base64Encode(`${cleanedAndFormattedArrayData}`);`

to this

`const encodedArrayData = Utilities.base64Encode(cleanedAndFormattedArrayData, Utilities.Charset.UTF_8);`

The encoding of the data was the problem. The second parameter tells the function what the encoding of the input is, and it seems that the google sheets data is in UTF-8. The google documentation doesn't state what charset is used when you don't use the second parameter.
